linux-kernel-dom0 build fails in fedora-33 with invalid `%P` in specfile

### Qubes OS release

4.1beta

### Brief summary

`make qubes-dom0` fails on linux-kernel-dom0, with the following error log:

```
$ cat build-logs/linux-kernel-dom0-fc32.log
make[2]: Entering directory '/home/user/qubes-builder'
/home/user/qubes-builder/qubes-src/builder-rpm/scripts/generate-spec /home/user/qubes-builder/chroot-dom0-fc32/home/user/qubes-src/linux-kernel/kernel.spec.in /home/user/qubes-builder/chroot-dom0-fc32/home/user/qubes-src/linux-kernel/kernel.spec
/home/user/qubes-builder/qubes-src/builder-rpm//update-local-repo.sh fc32
sudo BACKEND_VMM=xen  dnf --installroot=/home/user/qubes-builder/chroot-dom0-fc32 --installroot=/home/user/qubes-builder/chroot-dom0-fc32 --refresh -y update
Qubes OS Builder Repository                     2.9 MB/s | 3.0 kB     00:00    
Qubes OS Builder Repository                     8.2 MB/s |  86 kB     00:00    
Fedora 32 - x86_64                               72 kB/s |  23 kB     00:00    
Fedora 32 openh264 (From Cisco) - x86_64         20 kB/s | 990  B     00:00    
Fedora 32 - x86_64 - Updates                    384 kB/s |  22 kB     00:00    
Dependencies resolved.
Nothing to do.
Complete!
sudo BACKEND_VMM=xen  chroot /home/user/qubes-builder/chroot-dom0-fc32 su -c 'cd "/home/user/qubes-src/linux-kernel"; rpmspec -P --define "debug_package %{nil}" --define "fedora 32" --define "dist .fc32" --define "fedora 32" --define "dist .fc32" kernel.spec > kernel.spec.parsed' - user
sudo BACKEND_VMM=xen  dnf builddep --spec --installroot=/home/user/qubes-builder/chroot-dom0-fc32 --installroot=/home/user/qubes-builder/chroot-dom0-fc32  -y /home/user/qubes-builder/chroot-dom0-fc32/home/user/qubes-src/linux-kernel/kernel.spec.parsed
Qubes OS Builder Repository                     2.9 MB/s | 3.0 kB     00:00    
RPM: error: /home/user/qubes-builder/chroot-dom0-fc32/home/user/qubes-src/linux-kernel/kernel.spec.parsed: line 252: %P: argument expected
Failed to open: '/home/user/qubes-builder/chroot-dom0-fc32/home/user/qubes-src/linux-kernel/kernel.spec.parsed', not a valid spec file: can't parse specfile

Error: Some packages could not be found.
make[2]: *** [/home/user/qubes-builder/qubes-src/builder-rpm/Makefile-legacy.rpmbuilder:52: dist-build-dep.spec] Error 1
make[2]: Leaving directory '/home/user/qubes-builder'
```

### Steps to reproduce

- clone a fedora-33-dev template from the standard fedora-33 template
- install in template the requested packages listed at https://www.qubes-os.org/doc/qubes-builder/ (`dnf install gnupg git createrepo rpm-build make wget rpmdevtools python3-sh dialog rpm-sign dpkg-dev debootstrap python3-pyyaml devscripts perl-Digest-MD5 perl-Digest-SHA`)
- install in template the additional packages that `make install-deps` would add (`dnf install createrepo_c debootstrap devscripts dpkg-dev python3-sh rpmdevtools`

- create an AppVM based on this template and open a shell in it
- run `sudo mount -o remount,dev /home` to avoid chroot-related error
- apply the https://www.qubes-os.org/doc/qubes-builder/ steps until `make get-sources`
- run `make qubes-dom0`

### Initial investigation

- `kernel.spec.parsed` is indeed empty in the end
- if I run by hand the line generating it:
  ```
  sudo BACKEND_VMM=xen  chroot /home/user/qubes-builder/chroot-dom0-fc32 su -c 'cd "/home/user/qubes-src/linux-kernel"; rpmspec -P --define "debug_package %{nil}" --define "fedora 32" --define "dist .fc32" --define "fedora 32" --define "dist .fc32" kernel.spec > kernel.spec.parsed' - user
  ```
  ... then it really gets contents, and as reported in later error line 252 does show as:
  ```
  find . ! -type d -printf '%P\n' > /home/user/rpmbuild/BUILD/kernel-5.10.71/obj-files
  ```

It would look like `kernel.spec.in` contents get interpreted twice, and that this `%P` needs to get double quoting, and the patch below seems to do the job.
I'm not really comfortable with this however, why would this fail here where it was apparently working fine in a fedora-32 build host, so I'm leaving this to people more versed in RPM builds.

```
diff --git a/kernel.spec.in b/kernel.spec.in
index 2225257..220245d 100644
--- a/kernel.spec.in
+++ b/kernel.spec.in
@@ -198,7 +198,7 @@ fi
 make clean $MAKE_ARGS
 
 rm -f source
-find . ! -type d -printf '%%P\n' > %my_builddir/obj-files
+find . ! -type d -printf '%%%%P\n' > %my_builddir/obj-files
 
 rm -rf %_builddir/dummy-psu
 tar -x -C %_builddir -zf %{SOURCE1}
```

--- a/Makefile-legacy.rpmbuilder
+++ b/Makefile-legacy.rpmbuilder
@@ -48,7 +48,7 @@ endif
 dist-build-dep: refresh-update.$(YUM) dist-build-dep$(suffix $(PACKAGE))
 
 dist-build-dep.spec:
-	sudo $(CHROOT_ENV) chroot $(CHROOT_DIR) su -c 'cd "$(DIST_SRC)"; rpmspec -P --define "debug_package %{nil}" $(RPM_QUERY_DEFINES) $(PACKAGE) > $(PACKAGE).parsed' - $(RUN_AS_USER)
+	sudo $(CHROOT_ENV) chroot $(CHROOT_DIR) su -c 'cd "$(DIST_SRC)" && sed 's:%%:%%%%:' $(PACKAGE) > $(PACKAGE).tmp && rpmspec -P --define "debug_package %{nil}" $(RPM_QUERY_DEFINES) $(PACKAGE).tmp > $(PACKAGE).parsed' - $(RUN_AS_USER)
 	sudo $(CHROOT_ENV) $(YUM_BUILDDEP) $(YUM_OPTS) $(RPM_BUILD_EXTRA_DEFINES) -y $(CHROOT_DIR)$(DIST_SRC)/$(PACKAGE).parsed
 
 dist-build-dep.rpm:

